File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.*;
class Solution {
    public int[] solution(int n) {
        List<Integer> list = new ArrayList<>();
        int cnt = 0;
        while(n != 1) {
            list.add(n);
            if(n%2==0) n /= 2;
            else n = 3 * n + 1;
        }
        
        list.add(n);
        int[] answer = list.stream().mapToInt(Integer::intValue).toArray();
        
        return answer;
    }
}

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int n) {
        List<int> numbers = new List<int>();
        numbers.Add(n);
        
        while(n != 1)
        {
            n = n % 2 == 0 ? n / 2 : n * 3 + 1;
            numbers.Add(n);
        }
        int[] answer = numbers.ToArray();
        return answer;
    }
}